core/src/main/java/greencity/controller/UserAgreementController.java

```diff
@@ -0,0 +1,130 @@
+package greencity.controller;
+
+import greencity.constants.HttpStatuses;
+import greencity.dto.useragreement.UserAgreementDetailDto;
+import greencity.dto.useragreement.UserAgreementDto;
+import greencity.service.ubs.UserAgreementService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import java.security.Principal;
+import java.util.List;
+import static greencity.constant.AppConstant.USER_AGREEMENT_LINK;
+
+@RestController
+@RequestMapping(USER_AGREEMENT_LINK)
+@AllArgsConstructor
+public class UserAgreementController {
+    private final UserAgreementService userAgreementService;
+
+    /**
+     * Retrieves all user agreement IDs sorted by creation date in ascending order.
+     *
+     * @return a {@link ResponseEntity} containing a list of user agreement IDs
+     *         sorted from oldest to newest, with an HTTP status of 200 (OK).
```

Comment:
I reckon that 'with an HTTP status of 200 (OK)' could be removed because we can return 401 and 403 response codes too.